### ISSUE TITLE
Problem with external user commands due to backslashes in Windows

### ISSUE THREAD

**Issue by chopsueysensei** [ctrlpvim/ctrlp.vim#342]
Hi.
When using some external tools as 'ctrlp_user_command' in Windows (ripgrep specifically, but there were others) I always get problems derived from the fact that CtrlP sends the path argument (%s) with backslashes. In the case of ripgrep, the tool seems to assume that the backslashes are regex markers and fails, but I've seen similar problems with other tools that simply aren't prepared to use backslashes..

This in itself is not a problem with CtrlP, of course. The problem is that there's no way to solve this behaviour, because even setting 'shellslash' to on, CtrlP insists on sending backslashes.
So currently there's no way of using these tools on Windows without resorting to horrible hacks.
I thinkk at least the 'shellslash' setting should be honored..

**Comment by wontoncc** [comment]
I agree with you. I have encountered similar problem while using CtrlP with (ripgrep)[https://github.com/BurntSushi/ripgrep] and eventually made a dirty workaround in `autoload/ctrlp.vim`.

**Comment by chopsueysensei** [comment]
Well, my workaround is even worse since it involves an external batch file.. (I'm not at all versed in vimscript yet). Would you mind sharing how you solved this without leaving vim?

**Comment by wontoncc** [comment]
@chopsueysensei Sure, and sorry for replying this late. But I think my solution should disappoint you, since I accidentally found that `rg` works just fine with slashes on Windows. So I just replace the backslashes with slashes to avoid the regex problem:
``` Vim
" autoload/ctrlp.vim

" if (has('win32') || has('win64')) && match(&shell, 'sh') != -1
if (has('win32') || has('win64')) && (executable('rg') || (match(&shell, 'sh') != -1))
	let path = tr(path, '\', '/')
en
```

In my opinion, a new path argument provided like `%ss` with backslashes or spaces escaped can be a better solution. May try to do it some time later.

**Comment by mattn** [comment]
If you set shellslash, you MUST change shell. Or your shell name doesn't contain `sh`?
